network/eventManager.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017 The Alvalor Authors
+//
+// This file is part of Alvalor.
+//
+// Alvalor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Alvalor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with Alvalor.  If not, see <http://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type eventManager interface {
+	Disconnected(addr string)
+	Connected(addr string)
+	Received(addr string, msg interface{})
```

Comment:
Still waiting on the error return from the functions in case it times out. That way we can log inside the handlers rather than spreading the logging too much into other places.

Reply:
Updated code to propagate error to the caller. By handler, did you mean client of event manager?